src/main/java/RedisSessionManagerT8.java

```diff
@@ -0,0 +1,319 @@
+        StandardSession standardSession = (StandardSession)session;
+        Hashtable<String, Object> attributes = new Hashtable();
+        for (Enumeration<String> enumerator = standardSession.getAttributeNames(); enumerator.hasMoreElements();) {
+            String key = enumerator.nextElement();
+            attributes.put(key, standardSession.getAttribute(key));
+        }
+        return attributes;
+    }
+
+    //------------------------------------------------------------------------------------------------------------------
+    // END Utils
+    //------------------------------------------------------------------------------------------------------------------
+
+
+    //------------------------------------------------------------------------------------------------------------------
+    // START REDIS
+    //------------------------------------------------------------------------------------------------------------------
+
+    /**
+     * Method to saved session in redis
+     * @param session
+     * @return
+     * @throws IOException
+     */
+    protected boolean saveSession(Session session) throws IOException {
+        boolean errorSaving = true;
+        //Get the session Id
+        String id = session.getId();
+        Hashtable<String, Object> metadata = getMetadata(session);
+        Hashtable<String, Object> attributes = getAttributes(session);
+        try{
+            //Save metadata to Redis
+            byte[] encodedMetadata = Base64.getEncoder().encode(SerializationUtils.serialize(metadata));
+            withRedis((Jedis jedis)-> jedis.set((id + REDIS_METADATA_KEY).getBytes(), encodedMetadata));
+
+            //Save Attributes to Redis
+            byte[] encodedAttributes = Base64.getEncoder().encode(SerializationUtils.serialize(attributes));
+            withRedis((Jedis jedis)-> jedis.setex((id + REDIS_ATTRIBUTES_KEY).getBytes(), session.getMaxInactiveInterval(), encodedAttributes));
+
+            errorSaving = false;
+        }catch (Exception e){
+            log.error("Error - Context: saveToRedis. Description: " + e.getMessage());
+        }
+        return errorSaving;
+    }
+
+    private interface RedisCallback<T> {
+        T apply(Jedis jedis) throws IOException;
+    }
+
+    private <T> T withRedis(RedisCallback<T> callback) throws IOException {
+        Objects.requireNonNull(redisConnectionPool, "redis connection pool must be initialized");
+        Objects.requireNonNull(callback, "callback needs to be provided");
+        try(Jedis jedis = redisConnectionPool.getResource()){
+            try {
+                T result = callback.apply(jedis);
+                redisConnectionPool.returnResourceObject(jedis);
+                return result;
+            } catch(IOException ioe) {
+                redisConnectionPool.returnBrokenResource(jedis);
+                throw ioe;
+            }
+        }catch(Exception e) {
+            throw new RuntimeException("Error working with Redis", e);
```

Comment:
This should be the only catch for handling Redis.  You need to move the `returnBrokenResource` to here.